docs/src/main/asciidoc/sql.adoc

```diff
@@ -83,17 +70,89 @@ NOTE: If you provide your own `spring.datasource.url`, it will be ignored, unles
+
+In order to use the Spring Boot Starters for Google Cloud SQL, the Google Cloud SQL API must be enabled in your GCP project.
+
+To do that, go to the https://console.cloud.google.com/apis/library[API library page] of the Google Cloud Console, search for "Cloud SQL API", click the first result and enable the API.
+
+NOTE: There are several similar "Cloud SQL" results.
+You must access the "Google Cloud SQL API" one and enable the API from there.
+
+=== Spring Boot Starter for Google Cloud SQL
+
+The Spring Boot Starters for Google Cloud SQL provide an auto-configured https://docs.oracle.com/javaee/7/api/javax/jms/ConnectionFactory.html[`ConnectionFactory`] bean.
+Coupled with Spring R2DBC it produces a
+https://docs.spring.io/spring-data/r2dbc/docs/current/reference/html/#r2dbc.core[`R2dbcEntityTemplate`] object bean that allows for operations such as querying and modifying a database.
+
+[source,java]
+----
+public Flux<String> listUsers() {
+  return template.select(User.class).all().map(user -> user.toString());
+}
+----
+
+You can rely on
+https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-sql.html#boot-features-connect-to-production-database[Spring Boot data source auto-configuration] to configure a `ConnectionFactory` bean.
+In other words, properties like the SQL username, `spring.r2dbc.username`, and password, `spring.r2dbc.password` can be used.
+There is also some configuration specific to Google Cloud SQL (see "Cloud SQL Configuration Properties" section below).
+
+|===
+| Property name | Description | Required | Default value
+| `spring.r2dbc.username` | Database username | No | MySQL: `root`; PostgreSQL: `postgres`
+| `spring.r2dbc.password` | Database password | No | `null`
+|===
+
+NOTE: If you provide your own `spring.r2dbc.url`, it will be ignored, unless you disable Cloud SQL auto-configuration for R2DBC with `spring.cloud.gcp.sql.enabled=false` or `spring.cloud.gcp.sql.r2dbc.enabled=false` .
+
+==== `ConnectionFactory` creation flow
+
+Spring Boot starter for Google Cloud SQL registers a `R2dbcCloudSqlEnvironmentPostProcessor` that provides a correctly formatted `spring.r2dbc.url` property to the environment based on the properties mentioned above.
+It also provides a default value for `spring.r2dbc.username`, which can be overridden.
+The starter also configures credentials for the R2DBC connection based on the properties below.
+
+The user properties and the properties provided by the `R2dbcCloudSqlEnvironmentPostProcessor` are then used by https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-sql.html[Spring Boot] to create the `ConnectionFactory`.
+
+Using the created `ConnectionFactory` in conjunction with Spring R2DBC provides you with a fully configured and operational `R2dbcEntityTemplate` object that you can use to interact with your SQL database.
+Similar to the JDBC support, you can connect to your database with as little as a database and instance names.
```

Comment:
I'd move this sentence to the top, so it's the second sentence right after "The Spring Boot Starters for Google Cloud SQL provide an auto-configured https://docs.oracle.com/javaee/7/api/javax/jms/ConnectionFactory.html[`ConnectionFactory`] bean."

This describes the value of the starter.